pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+
+	//TODO: In the future if we require a secret to access the PTS store
+	// then this call will need to pass in that key.
+	urlJSON, err := httpClient.Do(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error retrieving pod template spec: %v\n", err)
+		return
+	}
+	defer urlJSON.Body.Close()
+
+	if urlJSON.StatusCode != 200 {
+		fmt.Printf("Expected 200 got: %v\n", urlJSON.StatusCode)
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+
+	tempPTS := &api.PodTemplateSpec{}
+	err = json.NewDecoder(urlJSON.Body).Decode(tempPTS)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error decoding PTS JSON Body: %v\n", err)
+		return
+	}
+
+	//If the passed pod template spec doesn't have prior annotations
+	//then we have to call the below line. Determine if we need to check this
+	//or if we are assuming all pod template specs have prior annotations.
+	// tempPTS.Annotations = make(map[string]string)
+	tempPTS.Annotations["trafficHosts"] = tempJSON.TrafficHosts
+
+	template := extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name: tempJSON.DeploymentName,
+		},
+		Spec: extensions.DeploymentSpec{
+			Replicas: tempJSON.Replicas,
+			Selector: &unversioned.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": tempPTS.Labels["app"],
+				},
+			},
+			Template: *tempPTS,
+		},
+	}
+
+	//Create Deployment
+	dep, err := client.Deployments(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).Create(&template)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error creating deployment: %v\n", err)
+		return
+	}
+	js, err := json.Marshal(dep)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error marshalling deployment: %v\n", err)
+	}
+
+	w.WriteHeader(201)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(js)
+
+	fmt.Printf("Created Deployment: %v\n", dep.GetName())
+}
+
+//getDeployment returns a deployment matching the given environmentGroupID, environmentName, and deploymentName
+func getDeployment(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	depList, err := client.Deployments(pathVars["environmentGroupID"] + "-" + pathVars["environment"]).List(api.ListOptions{
+		LabelSelector: labels.Everything(),
+	})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error retrieving deployment list: %v\n", err)
+		return
+	}
+	for _, value := range depList.Items {
+		if value.GetName() == pathVars["deployment"] {
+			js, err := json.Marshal(value)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				fmt.Printf("Error marshalling deployment: %v\n", err)
+			}
+
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "application/json")
+			w.Write(js)
+			fmt.Printf("Got Deployment: %v\n", value.GetName())
```

Comment:
You should add a `break` to exit the `for` loop after you've found what you're looking for.  No need in processing things we know we aren't interested in.